src/main/java/org/opentripplanner/graph_builder/linking/SimpleStreetSplitter.java

```diff
@@ -96,30 +101,26 @@
 
     private static GeometryFactory geometryFactory = GeometryUtils.getGeometryFactory();
 
-    //If true edges are split and new edges are created (used when linking transit stops etc. during graph building)
-    //If false new temporary edges are created and no edges are deleted (Used when searching for origin/destination)
-    private final boolean destructiveSplitting;
```

Comment:
This seems like the core change, and quite a good one. The splitter itself is not configured to do one kind of splitting - each call must specify whether it's destructive or not. Can you confirm for any reviewers that this is the core idea?

Reply:
With the newly refactored code, the major purpose of this PR is to attempt to have only one StreetSplitter instantiated per graph. Since there's only one splitter per graph, the StreetSplitter must be able to accept an argument to indicate the type of street splitting that should occur.

Reply:
Just to clarify: the main purpose of the PR is still the same right, to resolve the bug #2758? Do you mean you've adopted a different approach to fixing that bug, which involves ensuring there is a single spatial index of discoverable edges (including split edges)?